### PR TITLE
upgrade halo2 ecosystem dependencies

### PR DESCRIPTION
**Pull request by hero78119**
### Description

- halo2 (v2023_04_20 -> v0.3.0)
- halo2wrong (v2023_04_20 -> v2024_01_31)
- snark-verifier (v2023_04_20 -> v2024_01_31)
- remove halo2curves patching and fetch from `crate-io`, which got latest release `0.6.0`

### Issue Link

https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1691

### Type of change
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: zkevm-circuits/src/mpt_circuit.rs [hunk 1]
```diff
@@ -807,7 +807,7 @@ mod tests {
                 println!("{} {:?}", idx, path);
                 let prover = MockProver::<Fr>::run(degree as u32, &circuit, vec![]).unwrap();
                 assert_eq!(prover.verify_at_rows(0..num_rows, 0..num_rows,), Ok(()));
-                // assert_eq!(prover.verify_par(), Ok(()));
+                // assert_eq!(prover.verify(), Ok(()));
                 // prover.assert_satisfied();
```
Comment (ChihChengLiang) [inline zkevm-circuits/src/mpt_circuit.rs:811]: These two are comments. We can remove them
```suggestion
```

**Review by ChihChengLiang — APPROVED** [review]
LGTM with 1 nitpick.

**Review by han0110 — APPROVED** [review]
LGTM!

File: zkevm-circuits/src/mpt_circuit.rs (hunk 1 above)
Comment (hero78119) [reply zkevm-circuits/src/mpt_circuit.rs:811]: I think its ok to leave them to be clean up later, since mpt feature still under development.

**Pull request merged**